From e888eb28275876fb0fc8f4a29e8da02f3225ecab Mon Sep 17 00:00:00 2001
From: rohithsharmaks <rohithsharmaks@apache.org>
Date: Thu, 7 Jan 2016 09:38:47 +0530
Subject: [PATCH 1055/2848] YARN-4393. Fix intermittent test failure for
 TestResourceLocalizationService#testFailedDirsResourceRelease
 (Varun Saxana via rohithsharmaks)

(cherry picked from commit 791c1639ae0b351e0bf0b2ecec854dc72ab07935)

Change-Id: Id3ccd9a8a620ab3c614d5323a8fafd6d6492c9db
---
 .../localizer/TestResourceLocalizationService.java |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/TestResourceLocalizationService.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/TestResourceLocalizationService.java
index c5fbddc..0ece9a6 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/TestResourceLocalizationService.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/TestResourceLocalizationService.java
@@ -2316,6 +2316,8 @@ public boolean matches(Object o) {
           new ApplicationLocalizationEvent(
             LocalizationEventType.DESTROY_APPLICATION_RESOURCES, app);
       spyService.handle(destroyApp);
+      // Waits for APPLICATION_RESOURCES_CLEANEDUP event to be handled.
+      dispatcher.await();
       verify(applicationBus).handle(argThat(matchesAppDestroy));
 
       // verify we got the right delete calls
-- 
1.7.9.5

